hibernate-core/src/test/java/org/hibernate/test/hql/joinedSubclass/JoinedSubclassSubQueryTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.hql.joinedSubclass;
+
+import java.util.List;
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import org.junit.Test;
+
+import org.hibernate.Session;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+
+/**
+ * @author Stephen Fikes
+ * @author Gail Badner
+ */
+public class JoinedSubclassSubQueryTest extends BaseCoreFunctionalTestCase {
+
```

Comment:
The following test fails because the constraint refers to `Employee#firstName`, which is in `Person` table, but the generated SQL does not include a join with `Person` in the subquery.